apps/bulk_data_migration/entry_images_v2/migrate.py

```diff
@@ -0,0 +1,82 @@
+from django.db.models import Q
+from django.conf import settings
+
+from utils.common import parse_number
+
+from lead.models import LeadPreviewImage
+from entry.models import Entry
+from gallery.models import File
+
+"""
+python3 manage.py bulk_migrate entry_images_v2
+"""
+
+FILE_API_PREFIX = '{protocol}://{domain}{url}'.format(
+    protocol=settings.HTTP_PROTOCOL,
+    domain=settings.DJANGO_API_HOST,
+    url='/file/',
+)
+S3_URL_PREFIX = f'https://{settings.AWS_STORAGE_BUCKET_NAME_MEDIA}.s3.amazonaws.com/{settings.MEDIAFILES_LOCATION}/'
+
+
+def _get_file_from_file_url(entry, string):
+    try:
+        fileid = parse_number(string.rstrip('/').split('/')[-1])
+    except IndexError:
+        return
+    return fileid and File.objects.filter(id=fileid).first()
+
+
+def _get_file_from_s3_url(entry, string):
+    try:
+        file_path = '/'.join(string.split('?')[0].split('/')[4:])
+    except IndexError:
+        return
```

Comment:
Is the use of `...[4:]` okay? Just to be safe, if things change, and we always get the `IndexError`.

Reply:
According to the data currently in production, this should work without any issue. Also, there is image_raw with the original value which we can use to analyze and fix remaining if found.